### DISCUSSION TITLE
Groupchat, enforce use of /gen

### DISCUSSION THREAD

**Discussion by inqoperated** [capslock/stable-diffusion-bot#29] [Ideas]
Atm the bot generates images by just typing the query without the /gen command. While i enjoy this for private chat it is no good for a group chat. 
And can you whitelist whole groupchats via the group number?

**Comment by capslock** [comment]
> Atm the bot generates images by just typing the query without the /gen command. While i enjoy this for private chat it is no good for a group chat.

You can do this using bot privacy settings. In BotFather when looking at settings for your bot, under `Group Privacy`, turn on `Privacy Mode`. Now in group chats, your bot will only respond to slash commands, and will generate images as normal or with slash commands in private chats.

> And can you whitelist whole groupchats via the group number?

Yes, if you provide a chat id instead of a user id in the `allowed_users` list, it should allow any user in the chat.
